codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -430,17 +430,22 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		reconcileEvent = rof.ObserveFinalizeKind(ctx, resource)
 	}
 
-	if !r.skipStatusUpdates {
-		// Synchronize the status.
-		if {{.equalitySemantic|raw}}.DeepEqual(original.Status, resource.Status) {
-			// If we didn't change anything then don't call updateStatus.
-			// This is important because the copy we loaded from the injectionInformer's
-			// cache may be stale and we don't want to overwrite a prior update
-			// to status with this stale state.
-		} else if !isLeader {
-			logger.Warn("Saw status changes when we aren't the leader!")
-			// TODO: Consider logging the diff at Debug?
-		} else if err = r.updateStatus(original, resource); err != nil {
+	// Synchronize the status.
+	switch {
```

Comment:
I would change the order of these to the following:

```
	case !isLeader:
	case r.skipStatusUpdates:
	case {{.equalitySemantic|raw}}.DeepEqual(original.Status, resource.Status):
```

Reply:
I had it that way originally thinking the deepEqual is computationally more expensive, but that actually changes the functionality.

Doing so would print: `logger.Warn("Saw status changes when we aren't the leader!")` for all non-leaders all the time and is expecting the equality check to happen first

Reply:
We should warn when folks are mutating status and shouldn't be.  If it is noisy, then good 😅

Reply:
Right but if we made scotts' suggestion it would warn even on no-change (which is not what we want)

Reply:
Right!

/approve